src/components/habitEditor/habitEditorComponents/HabitEditorFrequency.js

```diff
@@ -0,0 +1,118 @@
+import styled from "styled-components";
+
+const HabitEditorFrequency = ({
+	getPrevDate,
+	habitCheckList,
+	getPrevMonth,
+}) => {
+	const dateArr1D = [];
+	const arrayLen = 210;
+	const rowLength = arrayLen / 7;
+	let color = "#ddd";
+	let colorFlag = "g";
+
+	for (let i = 0; i < arrayLen; i++) {
+		habitCheckList.find((dateChecked) => dateChecked === getPrevDate(i))
+			? (colorFlag = "b")
+			: (colorFlag = "g");
+
+		dateArr1D.push(
+			getPrevDate(i).slice(
+				getPrevDate(i).indexOf("/") + 1,
+				getPrevDate(i).lastIndexOf("/")
+			) + colorFlag
+		);
+	}
```

Comment:
A lot of computation will happen here on each rerender. Can we do some optimization to make sure this only happens when actually rather than on each rerender

Reply:
removed the colorFlag=g else part